### PR TITLE
adds keep-alive-interval to repair QUIC transport config

### PR DESCRIPTION
**Pull request by behzadnouri**
#### Problem
QUIC connections timeout due to infrequent repair requests.

#### Summary of Changes
Added `keep_alive_interval` to transport config.

### CODE REVIEW

File: core/src/repair/quic_endpoint.rs [hunk 1]
```diff
@@ -46,7 +46,11 @@ const CONNECT_SERVER_NAME: &str = "solana-repair";
 const CLIENT_CHANNEL_BUFFER: usize = 1 << 14;
 const ROUTER_CHANNEL_BUFFER: usize = 64;
 const CONNECTION_CACHE_CAPACITY: usize = 3072;
+
+// Transport config.
+const KEEP_ALIVE_INTERVAL: Duration = Duration::from_secs(4);
 const MAX_CONCURRENT_BIDI_STREAMS: VarInt = VarInt::from_u32(512);
+const MAX_IDLE_TIMEOUT: Duration = Duration::from_secs(10);
```
Comment (jbiseda) [inline core/src/repair/quic_endpoint.rs:53]: I see definitions for these in `sdk/src/quic.rs`. Can you add comments about why new definitions are needed?

Comment (behzadnouri) [reply core/src/repair/quic_endpoint.rs:53]: The ones in sdk are used for tpu, which has very different properties than repair. In particular tpu packets are only forwarded to the upcoming leaders which we know in advance who they are, it is a uni-directional traffic, and has a very different load and latency tolerance. Repair is bi-directional request-response, peers are randomly sampled, etc.

I pretty much rather not to reuse the same parameters at this stage until this code is stable and repair over quic protocol is fully rolled out over mainnet. Until then reusing same parameters can introduce regressions on one protocol while trying to optimize another protocol.

Comment (lijunwangs) [reply core/src/repair/quic_endpoint.rs:53]: I think just add some code comments on the reasoning will help.

File: core/src/repair/quic_endpoint.rs [hunk 2]
```diff
@@ -191,11 +195,14 @@ fn new_client_config(cert: Certificate, key: PrivateKey) -> Result<ClientConfig,
 }
 
 fn new_transport_config() -> TransportConfig {
+    let max_idle_timeout = IdleTimeout::try_from(MAX_IDLE_TIMEOUT).unwrap();
     let mut config = TransportConfig::default();
     config
+        .datagram_receive_buffer_size(None)
```
Comment (lijunwangs) [inline core/src/repair/quic_endpoint.rs:201]: Comment this to disable datagrams?

File: core/src/repair/quic_endpoint.rs (hunk 1 above)
Comment (behzadnouri) [reply core/src/repair/quic_endpoint.rs:55]: done.

File: core/src/repair/quic_endpoint.rs (hunk 2 above)
Comment (behzadnouri) [reply core/src/repair/quic_endpoint.rs:208]: done.

**Review by lijunwangs — APPROVED** [review]
Looks good to me.

**Pull request merged**